rs/nns/governance/src/network_economics.rs

```diff
@@ -0,0 +1,263 @@
+    };
+
+    pub const DEFAULT_START_REDUCING_VOTING_POWER_AFTER_SECONDS: u64 = 6 * ONE_MONTH_SECONDS;
+    pub const DEFAULT_CLEAR_FOLLOWING_AFTER_SECONDS: u64 = ONE_MONTH_SECONDS;
+
+    pub fn with_default_values() -> Self {
+        Self::DEFAULT
+    }
+
+    /// Returns 1 if a neuron has refreshed (its voting power/following)
+    /// recently.
+    ///
+    /// Otherwise, if a neuron has not refreshed for >
+    /// start_reducing_voting_power_after_seconds, returns < 1 (but >= 0).
+    ///
+    /// Once a neuron has not refresehd for
+    /// start_reducing_voting_power_after_seconds +
+    /// clear_following_after_seconds, this returns 0.
+    ///
+    /// Between these two points, the decrease is linear.
+    pub fn deciding_voting_power_adjustment_factor(
+        &self,
+        time_since_last_voting_power_refreshed: Duration,
+    ) -> Decimal {
+        self.deciding_voting_power_adjustment_factor_function()
+            .apply(time_since_last_voting_power_refreshed.as_secs())
+            .clamp(Decimal::from(0), Decimal::from(1))
+    }
+
+    fn deciding_voting_power_adjustment_factor_function(&self) -> LinearMap {
+        let from_range = {
+            let begin = self.get_start_reducing_voting_power_after_seconds();
+            let end = begin + self.get_clear_following_after_seconds();
+
+            begin..end
+        };
+
+        #[allow(clippy::reversed_empty_ranges)]
+        let to_range = 1..0;
+
+        LinearMap::new(from_range, to_range)
+    }
+
+    pub fn get_start_reducing_voting_power_after_seconds(&self) -> u64 {
+        self.start_reducing_voting_power_after_seconds
+            .unwrap_or(Self::DEFAULT_START_REDUCING_VOTING_POWER_AFTER_SECONDS)
+    }
+
+    pub fn get_clear_following_after_seconds(&self) -> u64 {
+        self.clear_following_after_seconds
+            .unwrap_or(Self::DEFAULT_CLEAR_FOLLOWING_AFTER_SECONDS)
+    }
+}
+
+// Seems like there is probably some way we can make #[derive(...)] generate
+// implementations of this, because the hand-crafted implementations below are
+// pretty dang (if not completely) repetitive. OTOH, it would be a lot of work,
+// and it is not clear that we would use it many times.
+pub(crate) trait InheritFrom {
+    /// Returns a modified copy of self where fields containing 0 are replaced
+    /// with the value from base.
+    fn inherit_from(&self, base: &Self) -> Self;
+}
+
+// Ideally, we'd use num_traits::Zero to give a generic implementation that
+// applies to all integer types, but Rust refuses to allow that in the presence
+// of impl InheritFrom for Option<T> below. If only there was some way we could
+// tell Rust, "ok, but this impl does not apply when the type happens to be
+// Option", but that doesn't exist (yet). Fortunately, we do not use a wide
+// range of integer types. Therefore, only this one is needed (for now).
+impl InheritFrom for u64 {
+    fn inherit_from(&self, base: &Self) -> Self {
+        if self == &0_u64 {
```

Comment:
Could this be done generically as follows?

```suggestion
        if self == &Self::default() {
```

and this code would be under 

```
impl<T: Default> InheritFrom for T { ... }
```